diff -urN _vendor/google-orig/auth/app_engine.py _vendor/google/auth/app_engine.py
--- _vendor/google-orig/auth/app_engine.py	2020-09-16 00:48:45.000000000 +0000
+++ _vendor/google/auth/app_engine.py	2021-05-07 21:24:21.859819512 +0000
@@ -24,9 +24,9 @@
 
 import datetime
 
-from google.auth import _helpers
-from google.auth import credentials
-from google.auth import crypt
+from artifact_registry._vendor.google.auth import _helpers
+from artifact_registry._vendor.google.auth import credentials
+from artifact_registry._vendor.google.auth import crypt
 
 # pytype: disable=import-error
 try:
@@ -39,7 +39,7 @@
 class Signer(crypt.Signer):
     """Signs messages using the App Engine App Identity service.
 
-    This can be used in place of :class:`google.auth.crypt.Signer` when
+    This can be used in place of :class:`artifact_registry._vendor.google.auth.crypt.Signer` when
     running in the App Engine standard environment.
     """
 
diff -urN _vendor/google-orig/auth/compute_engine/credentials.py _vendor/google/auth/compute_engine/credentials.py
--- _vendor/google-orig/auth/compute_engine/credentials.py	2020-09-16 00:48:45.000000000 +0000
+++ _vendor/google/auth/compute_engine/credentials.py	2021-05-07 21:24:21.822816728 +0000
@@ -29,7 +29,7 @@
 from google.auth import iam
 from google.auth import jwt
 from google.auth.compute_engine import _metadata
-from google.oauth2 import _client
+from artifact_registry._vendor.google.oauth2 import _client
 
 
 class Credentials(credentials.ReadOnlyScoped, credentials.CredentialsWithQuotaProject):
diff -urN _vendor/google-orig/auth/credentials.py _vendor/google/auth/credentials.py
--- _vendor/google-orig/auth/credentials.py	2020-09-16 00:48:45.000000000 +0000
+++ _vendor/google/auth/credentials.py	2021-05-07 21:24:21.861819662 +0000
@@ -145,7 +145,7 @@
                 billing purposes
 
         Returns:
-            google.oauth2.credentials.Credentials: A new credentials instance.
+            artifact_registry._vendor.google.oauth2.credentials.Credentials: A new credentials instance.
         """
         raise NotImplementedError("This credential does not support quota project.")
 
diff -urN _vendor/google-orig/auth/_default.py _vendor/google/auth/_default.py
--- _vendor/google-orig/auth/_default.py	2020-09-16 00:48:45.000000000 +0000
+++ _vendor/google/auth/_default.py	2021-05-07 21:24:21.852818985 +0000
@@ -111,7 +111,7 @@
     credential_type = info.get("type")
 
     if credential_type == _AUTHORIZED_USER_TYPE:
-        from google.oauth2 import credentials
+        from artifact_registry._vendor.google.oauth2 import credentials
 
         try:
             credentials = credentials.Credentials.from_authorized_user_info(
@@ -128,7 +128,7 @@
         return credentials, None
 
     elif credential_type == _SERVICE_ACCOUNT_TYPE:
-        from google.oauth2 import service_account
+        from artifact_registry._vendor.google.oauth2 import service_account
 
         try:
             credentials = service_account.Credentials.from_service_account_info(
diff -urN _vendor/google-orig/auth/impersonated_credentials.py _vendor/google/auth/impersonated_credentials.py
--- _vendor/google-orig/auth/impersonated_credentials.py	2020-09-16 00:48:45.000000000 +0000
+++ _vendor/google/auth/impersonated_credentials.py	2021-05-07 21:24:21.867820113 +0000
@@ -148,7 +148,7 @@
     Initialize a source credential which does not have access to
     list bucket::
 
-        from google.oauth2 import service_acccount
+        from artifact_registry._vendor.google.oauth2 import service_acccount
 
         target_scopes = [
             'https://www.googleapis.com/auth/devstorage.read_only']
diff -urN _vendor/google-orig/auth/_oauth2client.py _vendor/google/auth/_oauth2client.py
--- _vendor/google-orig/auth/_oauth2client.py	2020-09-16 00:48:45.000000000 +0000
+++ _vendor/google/auth/_oauth2client.py	2021-05-07 21:24:21.855819211 +0000
@@ -26,8 +26,8 @@
 from google.auth import _helpers
 import google.auth.app_engine
 import google.auth.compute_engine
-import google.oauth2.credentials
-import google.oauth2.service_account
+import artifact_registry._vendor.google.oauth2.credentials
+import artifact_registry._vendor.google.oauth2.service_account
 
 try:
     import oauth2client.client
@@ -48,7 +48,7 @@
 
 
 def _convert_oauth2_credentials(credentials):
-    """Converts to :class:`google.oauth2.credentials.Credentials`.
+    """Converts to :class:`artifact_registry._vendor.google.oauth2.credentials.Credentials`.
 
     Args:
         credentials (Union[oauth2client.client.OAuth2Credentials,
@@ -56,9 +56,9 @@
             convert.
 
     Returns:
-        google.oauth2.credentials.Credentials: The converted credentials.
+        artifact_registry._vendor.google.oauth2.credentials.Credentials: The converted credentials.
     """
-    new_credentials = google.oauth2.credentials.Credentials(
+    new_credentials = artifact_registry._vendor.google.oauth2.credentials.Credentials(
         token=credentials.access_token,
         refresh_token=credentials.refresh_token,
         token_uri=credentials.token_uri,
@@ -73,7 +73,7 @@
 
 
 def _convert_service_account_credentials(credentials):
-    """Converts to :class:`google.oauth2.service_account.Credentials`.
+    """Converts to :class:`artifact_registry._vendor.google.oauth2.service_account.Credentials`.
 
     Args:
         credentials (Union[
@@ -82,11 +82,11 @@
             credentials to convert.
 
     Returns:
-        google.oauth2.service_account.Credentials: The converted credentials.
+        artifact_registry._vendor.google.oauth2.service_account.Credentials: The converted credentials.
     """
     info = credentials.serialization_data.copy()
     info["token_uri"] = credentials.token_uri
-    return google.oauth2.service_account.Credentials.from_service_account_info(info)
+    return artifact_registry._vendor.google.oauth2.service_account.Credentials.from_service_account_info(info)
 
 
 def _convert_gce_app_assertion_credentials(credentials):
@@ -97,7 +97,7 @@
             credentials to convert.
 
     Returns:
-        google.oauth2.service_account.Credentials: The converted credentials.
+        artifact_registry._vendor.google.oauth2.service_account.Credentials: The converted credentials.
     """
     return google.auth.compute_engine.Credentials(
         service_account_email=credentials.service_account_email
@@ -112,7 +112,7 @@
             The credentials to convert.
 
     Returns:
-        google.oauth2.service_account.Credentials: The converted credentials.
+        artifact_registry._vendor.google.oauth2.service_account.Credentials: The converted credentials.
     """
     # pylint: disable=invalid-name
     return google.auth.app_engine.Credentials(
@@ -141,13 +141,13 @@
     This class converts:
 
     - :class:`oauth2client.client.OAuth2Credentials` to
-      :class:`google.oauth2.credentials.Credentials`.
+      :class:`artifact_registry._vendor.google.oauth2.credentials.Credentials`.
     - :class:`oauth2client.client.GoogleCredentials` to
-      :class:`google.oauth2.credentials.Credentials`.
+      :class:`artifact_registry._vendor.google.oauth2.credentials.Credentials`.
     - :class:`oauth2client.service_account.ServiceAccountCredentials` to
-      :class:`google.oauth2.service_account.Credentials`.
+      :class:`artifact_registry._vendor.google.oauth2.service_account.Credentials`.
     - :class:`oauth2client.service_account._JWTAccessCredentials` to
-      :class:`google.oauth2.service_account.Credentials`.
+      :class:`artifact_registry._vendor.google.oauth2.service_account.Credentials`.
     - :class:`oauth2client.contrib.gce.AppAssertionCredentials` to
       :class:`google.auth.compute_engine.Credentials`.
     - :class:`oauth2client.contrib.appengine.AppAssertionCredentials` to
diff -urN _vendor/google-orig/oauth2/credentials.py _vendor/google/oauth2/credentials.py
--- _vendor/google-orig/oauth2/credentials.py	2020-09-16 00:48:45.000000000 +0000
+++ _vendor/google/oauth2/credentials.py	2021-05-07 21:24:21.875820715 +0000
@@ -40,7 +40,7 @@
 from google.auth import _helpers
 from google.auth import credentials
 from google.auth import exceptions
-from google.oauth2 import _client
+from artifact_registry._vendor.google.oauth2 import _client
 
 
 # The Google OAuth 2.0 token endpoint. Used for authorized user credentials.
@@ -141,7 +141,7 @@
         Depending on the authorization server and the scopes requested, this
         may be populated when credentials are obtained and updated when
         :meth:`refresh` is called. This token is a JWT. It can be verified
-        and decoded using :func:`google.oauth2.id_token.verify_oauth2_token`.
+        and decoded using :func:`artifact_registry._vendor.google.oauth2.id_token.verify_oauth2_token`.
         """
         return self._id_token
 
@@ -226,7 +226,7 @@
                 credentials.
 
         Returns:
-            google.oauth2.credentials.Credentials: The constructed
+            artifact_registry._vendor.google.oauth2.credentials.Credentials: The constructed
                 credentials.
 
         Raises:
@@ -263,7 +263,7 @@
                 credentials.
 
         Returns:
-            google.oauth2.credentials.Credentials: The constructed
+            artifact_registry._vendor.google.oauth2.credentials.Credentials: The constructed
                 credentials.
 
         Raises:
@@ -331,7 +331,7 @@
             account (str): Account to get the access token for.
 
         Returns:
-            google.oauth2.credentials.UserAccessTokenCredentials: The created
+            artifact_registry._vendor.google.oauth2.credentials.UserAccessTokenCredentials: The created
                 credentials with the given account.
         """
         return self.__class__(account=account, quota_project_id=self._quota_project_id)
diff -urN _vendor/google-orig/oauth2/id_token.py _vendor/google/oauth2/id_token.py
--- _vendor/google-orig/oauth2/id_token.py	2020-09-16 00:48:45.000000000 +0000
+++ _vendor/google/oauth2/id_token.py	2021-05-07 21:24:21.877820866 +0000
@@ -25,7 +25,7 @@
 
 Example::
 
-    from google.oauth2 import id_token
+    from artifact_registry._vendor.google.oauth2 import id_token
     from google.auth.transport import requests
 
     request = requests.Request()
@@ -195,13 +195,13 @@
 
     Example::
 
-        import google.oauth2.id_token
+        import artifact_registry._vendor.google.oauth2.id_token
         import google.auth.transport.requests
 
         request = google.auth.transport.requests.Request()
         target_audience = "https://pubsub.googleapis.com"
 
-        id_token = google.oauth2.id_token.fetch_id_token(request, target_audience)
+        id_token = artifact_registry._vendor.google.oauth2.id_token.fetch_id_token(request, target_audience)
 
     Args:
         request (google.auth.transport.Request): A callable used to make
@@ -250,7 +250,7 @@
                 (info.get("type") == "service_account") and info or None
             )
 
-            from google.oauth2 import service_account
+            from artifact_registry._vendor.google.oauth2 import service_account
 
             credentials = service_account.IDTokenCredentials.from_service_account_info(
                 credentials_content, target_audience=audience
diff -urN _vendor/google-orig/oauth2/service_account.py _vendor/google/oauth2/service_account.py
--- _vendor/google-orig/oauth2/service_account.py	2020-09-16 00:48:45.000000000 +0000
+++ _vendor/google/oauth2/service_account.py	2021-05-07 21:24:21.880821092 +0000
@@ -77,7 +77,7 @@
 from google.auth import _service_account_info
 from google.auth import credentials
 from google.auth import jwt
-from google.oauth2 import _client
+from artifact_registry._vendor.google.oauth2 import _client
 
 _DEFAULT_TOKEN_LIFETIME_SECS = 3600  # 1 hour in seconds
 
